apps/prairielearn/src/server.js

```diff
@@ -1827,6 +1827,37 @@ module.exports.initExpress = function () {
     require('./pages/publicQuestions/publicQuestions'),
   ]);
 
+  // Client files for questions
+  app.use('/pl/public/course/:course_id/question/:question_id/clientFilesCourse', [
+    function (req, res, next) {
+      res.locals.public = true;
+      next();
+    },
+    require('./pages/clientFilesCourse/clientFilesCourse'),
+  ]);
+  app.use('/pl/public/course/:course_id/question/:question_id/clientFilesQuestion', [
+    function (req, res, next) {
+      res.locals.public = true;
+      next();
+    },
+    require('./pages/clientFilesQuestion/clientFilesQuestion'),
+  ]);
+
+  // generatedFiles
+  app.use('/pl/public/course/:course_id/question/:question_id/generatedFilesQuestion', [
+    function (req, res, next) {
+      res.locals.public = true;
+      next();
+    },
+    require('./pages/generatedFilesQuestion/generatedFilesQuestion'),
+  ]);
+
+  // TODO: implement
+  // // Submission files
+  // app.use('/pl/course/:course_id/question/:question_id/submission/:submission_id/file', [
+  //   require('./pages/submissionFile/submissionFile'),
+  // ]);
```

Comment:
Do we actually ever need to implement this endpoint for public question previews? Maybe it doesn't matter if the uploads to a public preview just disappear since they are not linked to a course or anything.

Reply:
I think we should; this is necessary for thing like `pl-file-preview` to work correctly.

Reply:
sounds good